tests/framework/extensions/clusters/eks/nodepools.go

```diff
@@ -0,0 +1,72 @@
+package eks
+
+import (
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/sirupsen/logrus"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	active = "active"
+)
+
+// updateNodePoolQuantity is a helper method that will update the node pool with the desired quantity.
+func updateNodePoolQuantity(client *rancher.Client, cluster *management.Cluster, nodePool *NodeGroupConfig) (*management.Cluster, error) {
+	clusterResp, err := client.Management.Cluster.ByID(cluster.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var eksConfig = clusterResp.EKSConfig
+	*eksConfig.NodeGroups[0].DesiredSize += *nodePool.DesiredSize
```

Comment:
So could we get an explanation on why we're indexing from zero. For our other scaling tests we don't do this right?

Reply:
In the Rancher clusters, we utilize nodeRoles to be able to parse through. Hosted clusters do not have a concept of nodeRoles, so we can't have an exact 1:1 parity between the Rancher clusters and the hosted clusters.

Reply:
But the hosted clusters do have a notion of node pools sightly different because it's the config, but how can we be sure we're modifying the correct node pool? Like our other tests are testing scaling agent, etcd, and control plane. Here we don't have the same scaling being done, we're just scaling a single node pool by 1, not really the same validations being done. Honestly, I would be ok to take out the scaling portion of it out, and putting in a separate PR and have a little more design discussion around it. Similar to how our other node scaling tests work.

Reply:
ignore this